### PR TITLE
DragAndDropHelper: Reduce build dependencies and code bloat

### PR DESCRIPTION
**Pull request by uklotzde**
* Moved implementations from header into newly added .cpp file
* Fixed missing #include directives

### CODE REVIEW

**Comment by daschuer** [comment]
Than you for this useful PR.
The diff in GitHub is useless  
I have tried to using Meld to track down the changes between h and c files but it fails as well.
Do you see a chance to split it into two or more commits with a minimum effort? 

In some places the the double indent for broken lines is missing, probably a left over from the original.

**Comment by uklotzde** [comment]
No, I will not put more work into this. I just copied the code into the .cpp file and split the function arguments onto separate lines.

**Comment by uklotzde** [comment]
I omitted to reformat the files. Either let me reformat the files with clang-format or otherwise inconsistencies are expected.

**Comment by daschuer** [comment]
OK, I have skimmed over the Meld compare again and I think we can merge it if CI is ready.

**Comment by daschuer** [comment]
Use clang-format if you like or just leave it is.

**Comment by uklotzde** [comment]
Auto-formatting revealed another indentation issue that I didn't notice.

The latest config file from #2035 generates only a few modifications. When splitting all function arguments on separate lines all those lines are preserved.

**Comment by daschuer** [comment]
Thank you LGTM.

However this discovers an issue in the c-lang-format setting.
I will comment in line.

File: src/util/dnd.cpp [hunk 1]
```diff
@@ -143,8 +140,8 @@ bool DragAndDropHelper::allowLoadToPlayer(
     }
 
     return !isPlaying || pConfig->getValueString(
-            ConfigKey("[Controls]",
-                        "AllowTrackLoadToPlayingDeck")).toInt();
+                                        ConfigKey("[Controls]", "AllowTrackLoadToPlayingDeck"))
```
Comment (daschuer) [inline src/util/dnd.cpp]: What happens here? I recommend to switch to an easy indent rule for line brakes that does not depends on the braces in the previous line which is afaik clang-format default. This does not play well with refactoring tools, and may move broken lines far right, out of sight.
Just 4 spaces for normal indent and 8 spaces for line brakes.

File: src/util/dnd.cpp [hunk 2]
```diff
@@ -162,10 +159,10 @@ bool DragAndDropHelper::allowDeckCloneAttempt(
...
-                // prevent cloning to ourself
-                event.mimeData()->text() == group ||
-                // only allow clone from decks
-                !PlayerManager::isDeckGroup(event.mimeData()->text(), nullptr)) {
+        // prevent cloning to ourself
+        event.mimeData()->text() == group ||
+        // only allow clone from decks
+        !PlayerManager::isDeckGroup(event.mimeData()->text(), nullptr)) {
         return false;
     }
```
Comment (daschuer) [inline src/util/dnd.cpp]: Here is an other odd case. I think clang tries to match the opening brace of the if statement.
This case works well with original Google style where norm as indent is 2 and the if ( produces the desired double indent.

File: src/util/dnd.cpp [hunk 3]
```diff
@@ -193,7 +189,7 @@ QList<QFileInfo> DragAndDropHelper::dropEventFiles(
     qDebug() << "mimeData.hasText()" << mimeData.hasText() << mimeData.text();
 
     if (!mimeData.hasUrls() ||
-            (mimeData.hasText() && mimeData.text() == sourceIdentifier)) {
+        (mimeData.hasText() && mimeData.text() == sourceIdentifier)) {
         return QList<QFileInfo>();
     }
```
Comment (daschuer) [inline src/util/dnd.cpp]: Also here, you cannot distinguish the line break from  the conditional statement.
Is this a new setting in you format file? I did not remember these problems in the original auto format PR.

File: src/util/dnd.cpp (hunk 1 above)
Comment (uklotzde) [reply src/util/dnd.cpp]: With ColumnLimit set to 0 clang-format does not insert or remove line breaks and just applies the indentation rules. This has to be fixed manually.

File: src/util/dnd.cpp (hunk 2 above)
Comment (uklotzde) [reply src/util/dnd.cpp]: ContinuationIndentWidth is set to 8. I have no idea why this is formatted in a wrong way.

File: src/util/dnd.cpp (hunk 3 above)
Comment (uklotzde) [reply src/util/dnd.cpp]: Same here, indentation rules fail.

File: src/util/dnd.cpp (hunk 2 above)
Comment (uklotzde) [reply src/util/dnd.cpp]: This works as expected: AlignAfterOpenBracket: DontAlign

**Comment by uklotzde** [comment]
I have updated #2035 to fix the formatting issues.

**Comment by daschuer** [comment]
Nice. Thank you for keeping this polished. Just waiting for CI.

**Comment by daschuer** [comment]
LGTM

**Pull request merged**